janus-idp/backstage-showcase
# Feat(RBAC): Add authentication

## Description

Adds authentication to the showcase app. In preparation for the RBAC work, we will need some form of authentication to ensure that the permissions are correctly applied and access is granted. This PR adds three authentication providers in the form of GitHub, Guest, and Oauth2Proxy. There is also an additional logic that looks at the `auth.environment` variable within the app-config to determine which sign on providers will be available when running the showcase app.

## Which issue(s) does this PR fix

- Fixes #259 

## PR acceptance criteria

Please make sure that the following steps are complete:

- GitHub Actions are completed and successful
- Unit Tests are updated and passing
- E2E Tests are updated and passing
- Documentation is updated if necessary
- Add a screenshot if the change is UX/UI related

## How to test changes / Special notes to the reviewer
To enable the GitHub and Guest providers the following needs to be added to the configuration file
```yaml
auth:
  environment: development
  providers:
    github:
      development:
        clientId: ${GITHUB_APP_CLIENT_ID}
        clientSecret: ${GITHUB_APP_CLIENT_SECRET}
```

To enable the Oauth2Proxy provider, the following needs to be added to the configuration file
```yaml
auth:
  environment: production
  providers:
    github:
      production:
        clientId: ${GITHUB_APP_CLIENT_ID}
        clientSecret: ${GITHUB_APP_CLIENT_SECRET}
    oauth2Proxy: {}
```
- Notice: that we still need to include the github auth declaration as it is needed for the GitHub plugins
